### DISCUSSION TITLE
How do I get information about the files that have been modified?

### DISCUSSION THREAD

**Discussion by cool2528** [libvips/libvips#2616] [Q&A] [answered]
libvips Why does the process not exit and the read file information is always old? I use the following code to read the file's height and width information which is always unmodified.
```
        auto image = vips_image_new_from_file(folder_path.toStdString().c_str(), nullptr);
        if (image) {
            auto height = vips_image_get_height(image);
            auto width = vips_image_get_width(image);
            qDebug() << height << width;
            g_object_unref(image);
        }
```
If I change the size of the file I want to read externally, the height and width information I get here is still the old file information.

**Comment by jcupitt** [comment, marked as answer]
Hi @cool2528,

libvips caches operations, so if you call the same function with the same parameters you get the previous result. This can save a lot of time -- it's like common-subexpression elimination.

If you need to reuse source filenames, you should turn the cache off with eg.:

```C
vips_cache_set_max( 0 );
```

Meaning "cache zero operations".

**Reply by jcupitt** [reply]
Docs here fwiw: https://www.libvips.org/API/current/VipsOperation.html#vips-cache-set-max

**Comment by cool2528** [comment]
Thank you for your reply I so that it also works well Thank you

```
    auto image = vips_image_new_from_file(file_path.toStdString().c_str(),  "access", VIPS_ACCESS_LAST, nullptr)
```

**Reply by jcupitt** [reply]
Don't use `LAST` -- it's just there for bindings. `SEQUENTIAL` would work too, but you'll only be able to read from the image once.
